src/policy/markcompactspace.rs

```diff
@@ -52,6 +52,18 @@ impl<VM: VMBinding> SFT for MarkCompactSpace<VM> {
         Self::is_marked(object)
     }
 
+    fn pin_object(&self, _object: ObjectReference) -> bool {
+        unimplemented!()
```

Comment:
I think for mark compact, its behavior on the pinning should be the same as copy space (copying is compulsory).